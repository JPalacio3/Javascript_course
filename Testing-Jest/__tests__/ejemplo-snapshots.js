/* Métodos para validar Objetos por medio de snapshots :
En Jest, los snapshots son una forma de probar que la salida de una función o componente coincida con un valor esperado. Los snapshots capturan la salida de una prueba y la almacenan en un archivo. Luego, en ejecuciones posteriores de la prueba, Jest comparará la salida actual con la salida almacenada en el snapshot para verificar si ha habido cambios inesperados.
*/

const cliente = {
    nombre: 'Joel Palacio',
    balance: 500,
    tipo: 'premium'
}

describe( 'Testing al cliente', () => {
    test( 'Es Joel', () => {
        expect( cliente ).toMatchSnapshot();
    } );
} )
