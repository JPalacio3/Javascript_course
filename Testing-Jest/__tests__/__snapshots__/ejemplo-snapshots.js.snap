// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing al cliente Es Joel 1`] = `
{
  "balance": 500,
  "nombre": "Joel Palacio",
  "tipo": "premium",
}
`;
